apps/storefront/src/components/layout/B3Nav.tsx

```diff
@@ -148,16 +148,22 @@ export default function B3Nav({ closeSidebar }: B3NavProps) {
     const subsidiariesPermission = getSubsidiariesPermission(routes);
 
     if (selectCompanyHierarchyId) {
-      routes = routes.filter((route) =>
-        route?.subsidiariesCompanyKey
+      routes = routes.filter((route) => {
```

Comment:
🤔 do we need to set `routes`? it looks like we could just return in both cases:

```suggestion
      return routes.filter((route) => {
```